src/sentry/static/sentry/app/views/organizationGroupDetails/utils.jsx

```diff
@@ -62,3 +65,58 @@ export function getEventEnvironment(event) {
 
   return tag ? tag.value : null;
 }
+
+const SUBSCRIPTION_REASONS = {
+  commented: t(
+    "You're receiving workflow notifications because you have commented on this issue."
+  ),
+  assigned: t(
+    "You're receiving workflow notifications because you were assigned to this issue."
+  ),
+  bookmarked: t(
+    "You're receiving workflow notifications because you have bookmarked this issue."
+  ),
+  changed_status: t(
+    "You're receiving workflow notifications because you have changed the status of this issue."
+  ),
+  mentioned: t(
+    "You're receiving workflow notifications because you have been mentioned in this issue."
+  ),
+};
+
+/**
+ * @param {object} group
+ * @param {boolean} removeLinks add/remove links to subscription reasons text (default: false)
+ * @returns Reason for subscription
+ */
+export function getSubscriptionReason(group, removeLinks = false) {
+  if (group.isSubscribed) {
+    if (group.subscriptionDetails) {
+      const {reason} = group.subscriptionDetails;
+      if (reason === 'unknown') {
+        return t(
+          "You're receiving workflow notifications because you are subscribed to this issue."
+        );
+      }
+
+      if (SUBSCRIPTION_REASONS.hasOwnProperty(reason)) {
+        return SUBSCRIPTION_REASONS[reason];
+      }
+    }
+
+    return tct(
+      "You're receiving updates because you are [link:subscribed to workflow notifications] for this project.",
+      {
+        link: removeLinks ? <span /> : <a href="/account/settings/notifications/" />,
+      }
+    );
+  }
+
+  if (group.subscriptionDetails && group.subscriptionDetails.disabled) {
+    return tct('You have [link:disabled workflow notifications] for this project.', {
+      link: removeLinks ? <span /> : <a href="/account/settings/notifications/" />,
+    });
+  }
+
+  return t("You're not subscribed to this issue.");
```

Comment:
nit: I might change the order of this so that the branches are: 1) group.subscriptDetails.disabled, 2) !group.isSubscribed 3) rest

this would remove a level of nesting for when user is subscribed.